include/pgduckdb/vendor/pg_fwd.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
```

Comment:
Ah good point, didn't think about proximity between `fdw` and `fwd` :-)

> Also, seems like this should be .hpp instead of .h.

What makes you say that? IMHO, since it is mostly C definitions, it should be a `.h` no?

Reply:
My assumption (which might be incorrect due to limited c++ knowledge) was that if a header uses C++ specific things, then it should be `hpp`. And if not, it can be `h`, i.e. if it can be included in a `c` file.

This one uses `#pragma once` and `extern 'C'`, both of which are not valid C constructs. So giving it `.hpp` seems better. 

And as a final benefit for our usage of splitting C and C++ code, I think marking this as `hpp` makes sense too. Because this should be imported by C++ code only. Postgres C code can just as well use the actual Postgres headers instead of this forward declaration stuff.

Reply:
> This one uses #pragma once and extern 'C', both of which are not valid C constructs. So giving it .hpp seems better.

So `#pragma once` is actually a non-official (but widely supported) C (and C++) preprocessor directive :-)
`extern 'C'` indeed is a shim for C++ and can be removed.

> And as a final benefit for our usage of splitting C and C++ code, I think marking this as hpp makes sense too. Because this should be imported by C++ code only. Postgres C code can just as well use the actual Postgres headers instead of this forward declaration stuff.

So while it is true that this file should only be imported by C++ files, it should by itself only contain C constructs (to be even more restrictive, it should actually only PG code).

I personally think that `.h` conveys a better idea of what it should contain but happy to change if you're not convinced :-)

Also would love to hear what @mkaruza's thoughts?

Reply:
> So #pragma once is actually a non-official (but widely supported) C (and C++) preprocessor directive :-)

Cool, I didn't realize this was something that worked in plain C too.

> `extern 'C'` indeed is a shim for C++ and can be removed.

To be clear I think it makes sense to keep `extern 'C'` in the header file here. That way the reviewer is triggered to review extra carefully whenever an `extern 'C'` is added in a C++ file.

> I personally think that .h conveys a better idea of what it should contain but happy to change if you're not convinced :-)

I'm not entirely convinced, but I also don't care a lot about it. Feel free to leave it like this, unless @mkaruza has a strong opinion about it.

Reply:
Sounds good - also sorry I replied on this comment, but the relevant PR is here https://github.com/duckdb/pg_duckdb/pull/386 :-)

Reply:
And if we're keeping the `extern 'C'` then I think I'm more convinced to change it to hpp. Will do that :-)

Reply:
Closing that convo that was pertaining to #386